runners/core-java/src/main/java/org/apache/beam/runners/core/EncodedBoundedWindow.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.core;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.io.ByteStreams;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.apache.beam.sdk.coders.AtomicCoder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.util.VarInt;
+import org.joda.time.Instant;
+
+/**
+ * An encoded {@link BoundedWindow} used within Runners to track window information without
+ * needing to decode the window.
+ *
+ * <p>This allows for Runners to not need to know window format during execution.
+ */
+@AutoValue
+public abstract class EncodedBoundedWindow extends BoundedWindow {
+  public static EncodedBoundedWindow forEncoding(ByteString encodedWindow) {
+    return new AutoValue_EncodedBoundedWindow(encodedWindow);
+  }
+
+  protected abstract ByteString getEncodedWindow();
+
+  public Instant maxTimestamp() {
+    throw new UnsupportedOperationException("TODO: Add support for reading the timestamp from "
```

Comment:
Wouldn't the point of this be to be a POJO with the max timestamp and the encoded window?

Reply:
After chatting, the confusion was around that this isn't meant to be extended and the protected visibility of the getEncodedWindow method is to support the usage of AutoValue.

Will change the visibility to public since it doesn't seem to matter.